drivers/pwm/pwm_ite_it8801.c

```diff
@@ -0,0 +1,183 @@
+	if (pulse_cycles == 0) {
+		duty = 0;
+	} else {
+		duty = pulse_cycles * 256 / period_cycles - 1;
+	}
+	LOG_DBG("IT8801 pwm duty cycles = %d", duty);
+
+	ret = i2c_reg_write_byte_dt(data->i2c_dev, config->reg_dcr, duty);
+	if (ret != 0) {
+		LOG_ERR("Failed to set cycles (ret %d)", ret);
+		return ret;
+	}
+
+	/* PWM channel clock source not gating */
+	pwm_enable(dev, 1);
+
+	return 0;
+}
+
+static int pwm_it8801_init(const struct device *dev)
+{
+	const struct pwm_it8801_config *config = dev->config;
+	struct pwm_it8801_data *data = dev->data;
+	int ret;
+
+	/* Verify multi-function parent is ready */
+	if (!device_is_ready(config->mfd)) {
+		LOG_ERR("(pwm)%s is not ready", config->mfd->name);
+		return -ENODEV;
+	}
+
+	data->i2c_dev = mfd_it8801_get_i2c_dt_spec(config->mfd);
+
+	/* PWM channel clock source gating before configuring */
+	pwm_enable(dev, 0);
+
+	for (int i = 0; i < config->mfdctrl_len; i++) {
+		/* Switching the pin to PWM alternate function */
+		ret = mfd_it8801_configure_pins(data->i2c_dev, config->altctrl[i].gpiocr,
+						config->altctrl[i].pin,
+						config->altctrl[i].alt_func);
+		if (ret != 0) {
+			LOG_ERR("Failed to configure pins (ret %d)", ret);
+			return ret;
+		}
+	}
+
+	return 0;
+}
+
+static const struct pwm_driver_api pwm_it8801_api = {
+	.set_cycles = pwm_it8801_set_cycles,
+	.get_cycles_per_sec = pwm_it8801_get_cycles_per_sec,
+};
+
+#define PWM_IT8801_INIT(inst)                                                                      \
+	static const struct it8801_mfd_pwm_altctrl_cfg                                             \
+		it8801_pwm_altctrl##inst[IT8801_DT_INST_MFDCTRL_LEN(inst)] =                       \
+			IT8801_DT_MFD_ITEMS_LIST(inst);                                            \
+	static const struct pwm_it8801_config pwm_it8801_cfg_##inst = {                            \
+		.mfd = DEVICE_DT_GET(DT_INST_PARENT(inst)),                                        \
+		.altctrl = it8801_pwm_altctrl##inst,                                               \
+		.mfdctrl_len = IT8801_DT_INST_MFDCTRL_LEN(inst),                                   \
+		.channel = DT_INST_PROP(inst, channel),                                            \
+		.reg_mcr = DT_INST_REG_ADDR_BY_IDX(inst, 0),                                       \
+		.reg_dcr = DT_INST_REG_ADDR_BY_IDX(inst, 1),                                       \
+		.reg_prslr = DT_INST_REG_ADDR_BY_IDX(inst, 2),                                     \
+		.reg_prsmr = DT_INST_REG_ADDR_BY_IDX(inst, 3),                                     \
+	};                                                                                         \
+	static struct pwm_it8801_data pwm_it8801_data_##inst;                                      \
+	DEVICE_DT_INST_DEFINE(inst, &pwm_it8801_init, NULL, &pwm_it8801_data_##inst,               \
+			      &pwm_it8801_cfg_##inst, POST_KERNEL,                                 \
+			      CONFIG_PWM_IT8801_INIT_PRIORITY, &pwm_it8801_api);
```

Comment:
Add a `BUILD_ASSERT` that the init priority is lower (higher number)  than the MFD parent.

Reply:
I suggested in a separate comment that this could all use the same priority, then the sub priority would take care of sequencing the init correctly, I think it's a better approach, less options kicking around. Build asserts should not be needed anymore since the build priorites are checked at build time btw.

Reply:
Is there anything I can refer to here? I'm encountering a build error when using CONFIG_MFD_INIT_PRIORITY + 1 in Kconfig or here.

Reply:
Use `CONFIG_MFD_INIT_PRIORITY `on all the child devices.